### PR TITLE
[expo-video] Add basic web implementation

### PR DESCRIPTION
**Pull request by aleqsio**
# Why

Adding web implementation to expo-video

# How

# Test Plan

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-video/src/VideoView.web.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
+import React, { useEffect, useRef, forwardRef, useImperativeHandle } from 'react';
+
+import { VideoPlayer } from './VideoPlayer';
+import { VideoViewProps } from './VideoView.types';
+class VideoPlayerImpl implements VideoPlayer {
```
Comment (tsapeta) [inline packages/expo-video/src/VideoView.web.tsx:5]:
```suggestion

class VideoPlayerImpl implements VideoPlayer {
```

File: packages/expo-video/src/VideoView.web.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    });
+  }
+  replay(): void {
+    this.mountedVideos.forEach((video) => {
+      video.currentTime = 0;
+      video.play();
+    });
+    this.isPlaying = true;
+  }
+  constructor(source: string | null = null) {
```
Comment (tsapeta) [inline packages/expo-video/src/VideoView.web.tsx:53]: constructor at the end of the class is something new to me 😂

File: packages/expo-video/src/VideoView.web.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  set isMuted(value: boolean) {
+    this.mountedVideos.forEach((video) => {
+      video.muted = value;
+    });
+    this._isMuted = value;
+  }
+  get isMuted(): boolean {
+    return this._isMuted;
+  }
+  timestamp: number = 0;
```
Comment (tsapeta) [inline packages/expo-video/src/VideoView.web.tsx:19]: Let's move members to the top

File: packages/expo-video/src/VideoView.web.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      }
+      videoRef.current?.requestFullscreen();
+    },
+    exitFullscreen: () => {
+      document.exitFullscreen();
+    },
+  }));
+
+  useEffect(() => {
+    if (!props.player || !videoRef.current) return;
```
Comment (tsapeta) [inline packages/expo-video/src/VideoView.web.tsx:80]:
```suggestion
    if (!props.player || !videoRef.current) {
      return;
    }
```

File: packages/expo-video/src/VideoView.web.tsx [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    props.player.mountedVideos.add(videoRef.current);
+    return () => {
+      if (videoRef.current) {
+        props.player?.mountedVideos.delete(videoRef.current);
+      }
+    };
+  }, [props.player]);
+  return (
```
Comment (tsapeta) [inline packages/expo-video/src/VideoView.web.tsx:88]:
```suggestion

  return (
```

File: packages/expo-video/src/VideoView.web.tsx [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    <video
+      {...props}
+      controls={props.nativeControls}
+      controlsList={props.allowsFullscreen ? undefined : 'nofullscreen'}
```
Comment (tsapeta) [inline packages/expo-video/src/VideoView.web.tsx:92]: Can't use `null`?

```suggestion
      controlsList={props.allowsFullscreen ? null : 'nofullscreen'}
```

File: packages/expo-video/src/VideoView.web.tsx [hunk 7]
```diff
@@ -0,0 +1,99 @@
+import React, { useEffect, useRef, forwardRef, useImperativeHandle } from 'react';
+
+import { VideoPlayer } from './VideoPlayer';
```
Comment (tsapeta) [inline packages/expo-video/src/VideoView.web.tsx:3]: We should probably move `VideoPlayer` class declaration to `VideoView.types.ts`

File: packages/expo-video/src/VideoView.web.tsx (hunk 1 above)
Comment (tsapeta) [reply packages/expo-video/src/VideoView.web.tsx:5]: Suggestion to consider: we could use `VideoPlayerWeb` name to indicate that this is an implementation for web

File: packages/expo-video/src/VideoView.types.ts [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+type VideoContentFit = 'contain' | 'cover' | 'fill';
+
+export type VideoViewProps = {
+  player: VideoPlayer;
+  nativeControls: boolean | undefined;
+  contentFit: VideoContentFit | undefined;
+  //   contentPosition: CGVector | undefined;
+  allowsFullscreen: boolean | undefined;
+  showsTimecodes: boolean | undefined;
+  requiresLinearPlayback: boolean | undefined;
```
Comment (tsapeta) [inline packages/expo-video/src/VideoView.types.ts:12]: Would be nice to add doc comments already, at least with the `@platform` tag so we can clearly see what is implemented on specific platform

File: packages/expo-video/src/VideoView.web.tsx (hunk 6 above)
Comment (aleqsio) [reply packages/expo-video/src/VideoView.web.tsx:92]: image
nope

**Review by tsapeta — APPROVED** [review]

**Pull request merged**